src/core/net/dnssd_server.cpp

```diff
@@ -431,21 +440,21 @@ Error Server::Response::ParseQueryName(void)
         uint8_t           labelLength = sizeof(label);
         uint16_t          comapreOffset;
 
-        SuccessOrExit(error = Name::ReadLabel(*mMessage, offset, label, labelLength));
-
-        if ((mQuestions.IsFor(kRrTypePtr) || mQuestions.IsFor(kRrTypeAny)) &&
-            StringMatch(label, kSubLabel, kStringCaseInsensitiveMatch))
-        {
-            mOffsets.mServiceName = offset;
-        }
-
         comapreOffset = offset;
 
         if (Name::CompareName(*mMessage, comapreOffset, kDefaultDomainName) == kErrorNone)
         {
             mOffsets.mDomainName = offset;
             ExitNow();
         }
+
+        SuccessOrExit(error = Name::ReadLabel(*mMessage, offset, label, labelLength));
```

Comment:
With this, it looks like line 460 is never reached. Could that be removed for clarity, or is there a reason to keep it?

Reply:
Removed it. Thanks.